zio-http/src/main/scala/zhttp/http/HttpData.scala

```diff
@@ -54,9 +54,9 @@ sealed trait HttpData[-R, +E] { self =>
    */
   private[zhttp] def unsafeSize: Long = self match {
     case HttpData.Empty               => 0L
-    case HttpData.Text(text, _)       => text.length.toLong
+    case HttpData.Text(text, charset) => text.getBytes(charset).size.toLong
     case HttpData.BinaryChunk(data)   => data.size.toLong
```

Comment:
Restored Text and BinaryChunk. As per test cases only one code path uses this. The Request => Response flow doesn't compute this.